### ISSUE TITLE
Urgent: Newly imported toolkits not appearing in search/filter

### ISSUE THREAD

**Issue by innovativeangela** [oecd-opsi/opsi#31]
I have been using Airtable to manage toolkit information. 

The previous developer working on the toolkit used a wordpress plugin called [All Import](https://wordpress.org/plugins/wp-all-import/) to import toolkit data into Wordpress to the Toolkit Navigator (post type: Toolkits). 

I ran an import today ([see ID 8](https://oecd-opsi.org/wp-admin/admin.php?page=pmxi-admin-manage)) Although new toolkits were added and are live on the site, only a small subset of toolkits show up in the filter view and via search function: 
https://oecd-opsi.org/search-toolkits/

The only toolkits that now appear were 7 uploaded previously (July 2018 and August 2018, according to wordpress dashboard filter). None of the recently added/updated Toolkits appear in search form, or on individual toolkit pages under "Other toolkits related to..." 
The new Toolkits do, however, show up on the right side of guide pages (for [example](https://oecd-opsi.org/guide/digital-transformation/))
Also, the "I have used this" and "Save toolkit" data are no longer linked between toolkits and user profiles. The import appears to have erased this data. 

Clues:
The import was configured prior to implementation of the ["Save toolkit"](https://github.com/oecd-opsi/meta-toolkit/issues/1) and ["I have used this toolkit"](https://github.com/oecd-opsi/meta-toolkit/issues/2) features, so perhaps there was a field mismatch.

**Comment by innovativeangela** [comment]
Jamie's suggestion:
1. You, though sFTP, remove the recaptcha requirement from a new AWS instance that Jamie created based on the snapshot of the site from yesterday. The URL for this is 35.176.6.48. 
2. You login to the dashboard and use the All Export plugin to explore a CSV for the Toolkit Navigator, including with the extra fields that may be creating the mismatch.
3. Import that CSV into the live site

**Comment by marcochiesi** [comment]
It seems that the import deleted some of the custom data for Toolkits, including the information for the ["Save toolkit"](https://github.com/oecd-opsi/meta-toolkit/issues/1) and ["I have used this toolkit"](https://github.com/oecd-opsi/meta-toolkit/issues/2), associations with taxonomy terms, the number of visits and the relevance.
In particular, the absence of the relevance is the cause of Toolkits disappearing on the frontend, because the relevance is the sort criteria when displaying the Toolkit search results. The relevance is recalculated once a day, so theoretically tomorrow all the Toolkits would have been visible again, but without all the other data.

That said, I restored ONLY the deleted data (meta fields and taxonomy associations for each toolkit) from the AWS instance that @jlberryhill created based on the snapshot from yesterday. That operation was made with low level and very secific SQL queries for exporting and reimporting only waht was needed. Otherwise, it would have been pretty complicated doing it with WP All Export as there are so many fields and taxonomies around that would have brought a high probability of errors.

But there's more, the images associated with the Toolkits were deleted and reimported again during the import. So it was necessary to maintain the images coming from today's import to avoid a more complex restore from yesterday's snapshot (complexity given by the need of restoring also the files and not just the DB data related to those images). Probably the Toolkit images did not change since last import, but you may want to double check that everything is ok.

In conclusion, the Toolkits existing on the site prior to today were restored to the status they had yesterday. The new Toolkits that were imported today were maintained on the system. 
So it should be an ideal scenario.

If you need or plan to perform more imports like this, I would recommend reconfigure WP All import so that it doesn't overwrite data, and also to perform some prelimianry tests on a staging environment.

**Comment by innovativeangela** [comment]
@marcochiesi  This seems to have resolved the issue. From now on, I will manually add new toolkits via Wordpress, instead of through the importer. If I need to import in bulk, I will have you help with the configuration of the plugin and/or test the import in staging.
